kernel/src/testing.rs

```diff
@@ -1,7 +1,65 @@
 use log::info;
 use test::ShouldPanic;
 
-use crate::{cpu::percpu::current_ghcb, sev::ghcb::GHCBIOSize};
+use crate::{
+    cpu::percpu::current_ghcb,
+    locking::{LockGuard, SpinLock},
+    serial::{SerialPort, Terminal},
+    sev::ghcb::GHCBIOSize,
+    svsm_console::SVSMIOPort,
+};
+
+use core::sync::atomic::{AtomicBool, Ordering};
+
+#[macro_export]
+macro_rules! assert_eq_warn {
+    ($left:expr, $right:expr) => {
+        if $left != $right {
+            log::warn!(
+                "Assertion warning failed at {}:{}:{}:\nassertion `left == rigth` failed\n left: {:?}\n right: {:?}",
+                file!(),
+                line!(),
+                column!(),
+                $left,
+                $right
+            );
+        }
+    };
+}
```

Comment:
This evaluates `$left` and `$right` twice in the error case. I suggest using temporary variables. There is also a typo in the error message ("rigth").

```suggestion
#[macro_export]
macro_rules! assert_eq_warn {
    ($left:expr, $right:expr) => {
        {
            let left = $left;
            let right = $right;
            if left != right {
                log::warn!(
                    "Assertion warning failed at {}:{}:{}:\nassertion `left == right` failed\n left: {left:?}\n right: {right:?}",
                    file!(),
                    line!(),
                    column!(),
                );
            }
        }
    };
}
```

Reply:
Applied, thanks!